minio/minio
# The prometheus metrics clean-up

## Description

The metrics are refactored and cleaned-up. We will expose the following metrics.

<b> minio_disk_* - labelled with the node address (Node specific) </b>

- minio_disk_storage_used_bytes 
- minio_disk_storage_available_bytes
- minio_disk_storage_total_bytes
  

<b>minio_http_* - Reads from the router level. labelled with node addr (node specific)</b>

- minio_http_rx_bytes_total
- minio_http_tx_bytes_total
- minio_http_s3_rx_bytes_total
- minio_http_s3_tx_bytes_total
- minio_http_s3_error_count
- minio_http_s3_request_count

<b> minio_disks_* - labelled with node addr </b>

- minio_disks_offline
- minio_disks_total

<b>minio_version_info - A gauge vector to show the commit and version </b>

- minio_version_info

<b>minio_http_requests_duration - A histogram vector to learn the latencies and method type-wise count</b> 

- minio_http_requests_duration_seconds_bucket (with latency and request type)


## Motivation and Context
Fixes #7873

## How to test this PR?
- Start the server
- visit /minio/prometheus/metrics to see the metrics

## Types of changes

## Checklist:
- [x] (Migration) Documentation needed
